src/v/cloud_topics/level_one/metastore/simple_metastore.cc

```diff
@@ -532,6 +533,15 @@ simple_metastore::get_compaction_offsets(
     }
     auto& prt = prt_ref->get();
     compaction_offsets_response resp;
+
+    resp.extents.reserve(prt.extents.size());
+    for (const auto& extent : prt.extents) {
+        resp.extents.push_back(
+          offset_interval_set::interval{
```

Comment:
It's surprising to see this be tied to the offset_interval_set, vs being some new standalone extent_bounds struct in the metastore / compaction_metadata interface. It's otherwise a bit misleading because the offset interval sets in the metastore generally don't refer to extents. Without a good reason to, I'd prefer keeping them separate